compiler/src/dotty/tools/backend/jvm/DottyBackendInterface.scala

```diff
@@ -796,11 +796,16 @@ class DottyBackendInterface(outputDirectory: AbstractFile, val superCallsMap: Ma
       for (f <- toDenot(sym).info.decls.toList if f.isMethod && f.isTerm && !f.isModule) yield f
     def serialVUID: Option[Long] =
       sym.getAnnotation(defn.SerialVersionUIDAnnot).flatMap { annot =>
-        val vuid = annot.argumentConstant(0).map(_.longValue)
-        if (vuid.isEmpty)
-          ctx.error("The argument passed to @SerialVersionUID must be a constant",
-            annot.argument(0).getOrElse(annot.tree).sourcePos)
-        vuid
+        if (sym.is(Flags.Trait)) {
+          ctx.error("@SerialVersionUID does nothing on a trait", annot.tree.sourcePos)
```

Comment:
I don't think it should be a hard error, or at least it looks like this is fine in java:

The problem is that dotty is giving `serialVersionUID` private access

```java
// Foo.java
public interface Foo {
  long serialVersionUID = 1L;
}
```
```
$ javap Foo
Compiled from "Foo.java"
public interface Foo {
  public static final long serialVersionUID;
}
```

Reply:
Fair enough, but it's still not something that means anything, see https://docs.oracle.com/en/java/javase/13/docs/api/java.base/java/io/Serializable.html. serialVersionUID is a thing you can use in a class that you serialize, but it doesn't do anything on an interface.

Reply:
although if the contract of the annotation is a private field then I will approve

Reply:
Ah yeah i see only useful if it is not inherited